File: src/test/testdata/inspections/unsafeIterators/Array.java
import java.util.Iterator;

import com.badlogic.gdx.utils.Array;
import com.badlogic.gdx.utils.Predicate;

class ArrayTest {

  class Pred implements Predicate<String> {
    public boolean evaluate(String arg0) {
      return false;
    }
  }

  void test() {
    Array<String> strings = new Array<String>();

    Iterator iterator1 = <warning>strings.iterator()</warning>;
    Iterable iterator2 = <warning>strings.select(new Pred())</warning>;

    for (String s: <warning>strings</warning>) {
      System.out.println(s);
    }
  }

}

File: src/test/testdata/inspections/unsafeIterators/Array.kt
import com.badlogic.gdx.utils.Array
import com.badlogic.gdx.utils.Predicate

@Suppress("UNUSED_VARIABLE")
class ArrayTest {

  inner class Pred : Predicate<String> {
    override fun evaluate(arg0: String): Boolean {
      return false
    }
  }

  fun test() {
    val strings = Array<String>()

    val iterator1: Iterator<Any> = <warning>strings.iterator()</warning>
    val iterator2: Iterable<Any> = <warning>strings.select(Pred())</warning>

    for (s in <warning>strings</warning>) println(s)
    for (s in <warning>Array<String>()</warning>) println(s)

    println(iterator1.toString())

  }

}